edit/base.js

```diff
@@ -48,6 +63,19 @@ const editor = Object.assign(EventEmitter(), {
       customName || name || t('styleMissingName')
     } - Stylus`; // the suffix enables external utilities to process our windows e.g. pin on top
   },
+};
+
+editor.on('styleUpdated', (newStyle, reason) => {
```

Comment:
I used `styleChange` intentionally to distinguish them from `styleUpdated`, especially as a reminder for us to avoid infinite loop. The event may be generate by `sections-editor.js`, `source-editor.js`, or `edit.js`.

Reply:
Using synonyms is wrong because its meaning is the same as the original event. Like we forgot about the first one and added another one. I don't understand why we need a reminder, but if we do, it should be an *explicit* one e.g. a comment on the event or an *entirely* different name.

Reply:
Hey, I think I kinda get it now so maybe we can differentiate by using a different way to invoke it e.g. a Proxy on `emit`...

Reply:
Another method is to avoid emitting `styleUpdated` from `sections-editor.js` and `source-editor.js`. Use a new event like `styleSaved`. Therefore `styleUpdated` is always emitted from `styleMan`.

We may not need the `styleSaved` event at all. We use that event to enable the setting page after a new style is created and saved. Because the settings page is opened via a button in this PR, we can just hide the button via CSS
```css
.is-new-style #style-cfg-btn {
  display: none;
}
```

Reply:
> we can just hide the button

Great idea! I've implemented it.